docs/SearchInput.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: default
+title: "The SearchInput Component"
+---
+
+# `<SearchInput>`
+
+```tsx
+import { SearchInput, TextInput, SelectInput } from 'react-admin';
+
+const postFilters = [
+    <SearchInput source="q" alwaysOn />,
+    <TextInput label="Title" source="title" defaultValue="Hello, World!" />,
+    <SelectInput source="category" choices={choices} />,
+];
+
+export const PostList = () => (
+    <List filters={postFilters}>
+        ...
+    </List>
+);
+```
+
+In addition to [the usual input types](./Inputs.md) (`<TextInput>`, `<SelectInput>`, `<ReferenceInput>`, etc.), you can use the `<SearchInput>` in the `filters` array. This input is designed especially for the [`Filter Form`](./FilterForm.md). It's like a `<TextInput resettable>` with a magnifier glass icon - exactly the type of input users look for when they want to do a full-text search.
+
+<video controls autoplay playsinline muted loop>
+  <source src="./img/search_input.webm" type="video/webm"/>
+  <source src="./img/search_input.mp4" type="video/mp4"/>
+  Your browser does not support the video tag.
+</video>
+
+Prefer using [`<FilterLiveSearch>`](./FilterLiveSearch.md) component if you want to provide your users a search features in a [`<List>` aside](./List.md#aside).
+
+<video controls autoplay playsinline muted loop>
+  <source src="./img/filter-live-search.webm" type="video/webm"/>
+  <source src="./img/filter-live-search.mp4" type="video/mp4"/>
+  Your browser does not support the video tag.
+</video>
+
+## Usage
+
+```jsx
+import { SearchInput, TextInput } from 'react-admin';
+
+const postFilters = [
+    <SearchInput source="q" alwaysOn />
+];
+```
+
+In the example given above, the `q` filter triggers a full-text search on all fields. It's your responsibility to implement the full-text filtering capabilities in your `dataProvider`, or in your API.
+
+## Props
+
+| Prop         | Required | Type      | Default | Description                                                          |
+| ------------ | -------- | --------- | ------- | -------------------------------------------------------------------- |
+| `placeholder`       | Optional | `string`  | `Search`  | Attribute for displaying default text in the `input` element |
+| `resettable`       | Optional | `boolean`  | `true`  | If `tue`, a delete button appear on input writing                       |
+| `variant`       | Optional | `string`  | `filled`  | Type attribute passed to the `<input>` element. Could be `filled` or `standard` too                       |
```

Comment:
`variant` is a standard MUI prop (not managed by react-admin). We don't even document it for `TextInput`, so I believe you can remove it from our docs (it is mentioned in the common input props section though).

```suggestion
```